### PR TITLE
Introduce a trivial cmake procedure and a Makefile for standalone application

### PR DESCRIPTION
**Pull request by latug0**
An environment setting file is generated during ` make install` .
A standalone application is put in the `install/examples` directory.
In the `README.md`, some instructions are given to use this feature.
The end user can compile easily both with cmake or make.

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+install(FILES
+  ex1/CMakeLists.txt
+  ex1/Makefile
+  DESTINATION share/examples/ex1)
```
Comment (thelfer) [inline examples/CMakeLists.txt:5]: share/mfem-mgis/examples/ex1 serait bien non ?

**Comment by thelfer** [comment]
Some remarks:

1. Please move `examples` in `share/mfem-mgis/`
2. It does not work on my computer on which `UMFPACK` is installed:

~~~~
$ make
/usr/bin/c++ -g -Wall -W -pedantic -isystem /tmp/install/share/mfem-mgis/cmake/../../../include -isystem /home/th202608/codes/mfem/master/install/lib/cmake/mfem//../../../include -isystem /include -isystem /include -isystem /home/th202608/codes/mgis/master/install-python-3.7/share/mgis/cmake//../../../include -O2 -std=c++17 UniaxialTensileTest.cxx -o UniaxialTensileTest -L/tmp/install/share/mfem-mgis/cmake/../../../lib -L/home/th202608/codes/mfem/master/install/lib/cmake/mfem//../../../lib -L/home/th202608/codes/mgis/master/install-python-3.7/share/mgis/cmake//../../../lib -L/lib -L/lib -lMFEMMGIS -lMFrontGenericInterface -lmfem -lHYPRE -lmetis 
In file included from UniaxialTensileTest.cxx:12:
/home/th202608/codes/mfem/master/install/include/mfem/linalg/solvers.hpp:24:10: fatal error: umfpack.h: Aucun fichier ou dossier de ce type
 #include <umfpack.h>
          ^~~~~~~~~~~
~~~~

**Comment by latug0** [comment]
Last remarks have been taken into account. If you can review one again this pull request please.

**Comment by thelfer** [comment]
Hi Guillaume,

I corrected the paths in `INSTALL.md`.

The `cmake` case is ok.

The `make` case isn't:

~~~~
$ make
/usr/bin/c++ -g -Wall -W -pedantic -I /tmp/install/share/mfem-mgis/cmake/../../../include -I /home/th202608/codes/mfem/master/install/lib/cmake/mfem//../../../include -I /home/th202608/codes/mgis/master/install-python-3.7/share/mgis/cmake//../../../include  -isystem /home/th202608/codes/mfem/master/install/include -isystem /usr/include -isystem /usr/include/suitesparse -O2 -std=c++17 UniaxialTensileTest.cxx -o UniaxialTensileTest -L/tmp/install/share/mfem-mgis/cmake/../../../lib -L/home/th202608/codes/mfem/master/install/lib/cmake/mfem//../../../lib -L/home/th202608/codes/mgis/master/install-python-3.7/share/mgis/cmake//../../../lib -lMFEMMGIS -lMFrontGenericInterface -lmfem  MFEM_LIB_LIST-NOTFOUND
c++: error: MFEM_LIB_LIST-NOTFOUND: Aucun fichier ou dossier de ce type
make: *** [Makefile:28: UniaxialTensileTest] Error 1
~~~~

**Comment by thelfer** [comment]
Concerning the make case, I think that we shall use the `config.mk` provided by `MFEM` as have a line similar to the following at the beggning of the `Makefile`:

~~~~
include $(MFEM_DIR)/share/mfem/config.mk
~~~~

**Comment by latug0** [comment]
> $ make
> /usr/bin/c++ -g -Wall -W -pedantic -I /tmp/install/share/mfem-mgis/cmake/../../../include -I /home/th202608/codes/mfem/master/install/lib/cmake/mfem//../../../include -I /home/th202608/codes/mgis/master/install-python-3.7/share/mgis/cmake//../../../include  -isystem /home/th202608/codes/mfem/master/install/include -isystem /usr/include -isystem /usr/include/suitesparse -O2 -std=c++17 UniaxialTensileTest.cxx -o UniaxialTensileTest -L/tmp/install/share/mfem-mgis/cmake/../../../lib -L/home/th202608/codes/mfem/master/install/lib/cmake/mfem//../../../lib -L/home/th202608/codes/mgis/master/install-python-3.7/share/mgis/cmake//../../../lib -lMFEMMGIS -lMFrontGenericInterface -lmfem  MFEM_LIB_LIST-NOTFOUND
> c++: error: MFEM_LIB_LIST-NOTFOUND: Aucun fichier ou dossier de ce type
> make: *** [Makefile:28: UniaxialTensileTest] Error 1

Can you give me the content of the command 
`env > myenv.txt`

The `source env.sh` should have defined the `MFEM_LIB_LIST` variable.

**Comment by latug0** [comment]
> Concerning the make case, I think that we shall use the `config.mk` provided by `MFEM` as have a line similar to the following at the beggning of the `Makefile`:
> 
> ```
> include $(MFEM_DIR)/share/mfem/config.mk
> ```

Good idea. I try your proposal.

**Comment by latug0** [comment]
> Concerning the make case, I think that we shall use the `config.mk` provided by `MFEM` as have a line similar to the following at the beggning of the `Makefile`:
> 
> ```
> include $(MFEM_DIR)/share/mfem/config.mk
> ```
The mast commit 094d80c use directly the mfem generated file `config.mk`.
Can you check that it is running well on your machines ?

**Comment by thelfer** [comment]
Here it is !
[env.zip](https://github.com/thelfer/mfem-mgis/files/6223434/env.zip)

**Comment by thelfer** [comment]
> Can you give me the content of the command
> `env > myenv.txt`
> 
> The `source env.sh` should have defined the `MFEM_LIB_LIST` variable.

Here it is !
[env.zip](https://github.com/thelfer/mfem-mgis/files/6223442/env.zip)

**Comment by thelfer** [comment]
> > Concerning the make case, I think that we shall use the `config.mk` provided by `MFEM` as have a line similar to the following at the beggning of the `Makefile`:
> > ```
> > include $(MFEM_DIR)/share/mfem/config.mk
> > ```
> 
> The mast commit [094d80c](https://github.com/thelfer/mfem-mgis/commit/094d80c78edf7e462eb9a78166c4b88548378b76) use directly the mfem generated file `config.mk`.
> Can you check that it is running well on your machines ?

Got a strange error...

~~~~
$ make
/usr/bin/c++ -isystem /home/th202608/codes/mfem/master/install/include -isystem /usr/include -isystem /usr/include/suitesparse -isystem /tmp/install/share/mfem-mgis/cmake/../../../include -isystem /home/th202608/codes/mgis/master/install-python-3.7/share/mgis/cmake//../../../include  -g -Wall -W -pedantic                     -O2 -std=c++17  UniaxialTensileTest.cxx -o UniaxialTensileTest -L/tmp/install/share/mfem-mgis/cmake/../../../lib -L/home/th202608/codes/mgis/master/install-python-3.7/share/mgis/cmake//../../../lib -lMFEMMGIS -lMFrontGenericInterface -L/home/th202608/codes/mfem/master/install/lib -lmfem -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lmetis -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lumfpack -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lklu -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lamd -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lbtf -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lcholmod -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lcolamd -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lcamd -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lccolamd -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lsuitesparseconfig -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lopenblas 
In file included from /usr/include/c++/8/ext/string_conversions.h:41,
                 from /usr/include/c++/8/bits/basic_string.h:6400,
                 from /usr/include/c++/8/string:52,
                 from /usr/include/c++/8/stdexcept:39,
                 from /usr/include/c++/8/array:39,
                 from /usr/include/c++/8/tuple:39,
                 from /usr/include/c++/8/bits/unique_ptr.h:37,
                 from /usr/include/c++/8/memory:80,
                 from UniaxialTensileTest.cxx:8:
/usr/include/c++/8/cstdlib:75:15: fatal error: stdlib.h: Aucun fichier ou dossier de ce type
 #include_next <stdlib.h>
               ^~~~~~~~~~
compilation terminated.
make: *** [Makefile:22: UniaxialTensileTest] Error 1
~~~~~

**Comment by thelfer** [comment]
> Got a strange error...

The error comes from the  `-isystem /usr/include` switch...

which itself comes from:

~~~~
MFEM_ISYSTEM := $(subst -I,-isystem ,$(MFEM_INCFLAGS))
~~~~

Replacing the latter by:

~~~~
MFEM_ISYSTEM := $(MFEM_INCFLAGS)
~~~~

works ! But with a ton of warnings !

**Comment by latug0** [comment]
Thanks for your answers.
The new fix should provide a workaround for the malfunctioning "-isystem /usr/include'

**Comment by thelfer** [comment]
Hi Guillaume,

Just make a few changes:

1. The `-g` flag is now only added if the DEBUG argument is passed to `make` in the `Makefile` case
2. The installation guide has been updated

I now merge the PR. Thanks for the contribution !

**Pull request merged**